src/classes/SRVC_Contact_PreventDeletion_TEST.cls

```diff
@@ -0,0 +1,310 @@
+        insert caseRecord;
+
+        Language__c language = UTIL_UnitTestData_TEST.createLanguage('Spanish');
+
+        Contact_Language__c contactLanguage = UTIL_UnitTestData_TEST.createContactLanguage(language.Id, contacts[0].Id, false);
+
+        Course_Offering__c courseOffering = UTIL_UnitTestData_TEST.createCourseOffering(course.Id, term.Id);
+        courseOffering.Faculty__c = contacts[0].Id;
+        update courseOffering;
+
+        Course_Enrollment__c courseCxn = UTIL_UnitTestData_TEST.getCourseConnection(contacts[0].Id, courseOffering.Id);
+        insert courseCxn;
+
+        Term_Grade__c termGrade = UTIL_UnitTestData_TEST.getTermGrade(contacts[0].Id,  courseOffering.Id, courseCxn.Id);
+        insert termGrade;
+
+        Program_Enrollment__c programEnrollment = UTIL_UnitTestData_TEST.getProgramEnrollment(acc.Id);
+        programEnrollment.Contact__c = contacts[0].Id;
+        insert programEnrollment;
+
+        Test__c testRecord = UTIL_UnitTestData_TEST.getTest('SAT', System.today(), contacts[0].Id);
+        insert testRecord;
+
+        Behavior_Involvement__c behInvolvement = UTIL_UnitTestData_TEST.getBehaviorInvolvementRecord(contacts[0].Id, caseRecord.Id, 'Reporter');
+        insert behInvolvement;
+
+    }
+
+    // Tests getInstance()
+    @isTest
+    static void testGetInstance() {
+
+        SRVC_Contact_PreventDeletion srvcConPreventDeleteInstance = SRVC_Contact_PreventDeletion.getInstance();
+        System.assertEquals(SRVC_Contact_PreventDeletion.instance, srvcConPreventDeleteInstance);
+        System.assertNotEquals(NULL, srvcConPreventDeleteInstance);
+
+        srvcConPreventDeleteInstance = SRVC_Contact_PreventDeletion.getInstance();
+        System.assertEquals(SRVC_Contact_PreventDeletion.instance, srvcConPreventDeleteInstance);
+    }
+
+    // Tests Prevent Deletion for Contacts with STUB_AlwaysHasChildRelationships
+    @isTest
+    static void testPreventDeletionForContactsWithChildren() {
+
+        SRVC_Contact_PreventDeletion.instance = new STUB_AlwaysHasChildRelationships();
+
+        Map<Id,Contact> contactById = new Map<Id,Contact>();
+
+        Test.startTest();
+            List<Contact> contactsWithErrorsList = SRVC_Contact_PreventDeletion.getInstance().PreventDeletionForContacts(
+                contactById
+            );
+        Test.stopTest();
+
+        System.assertEquals(contactById.values().size(), contactsWithErrorsList.size());
+
+        for (Contact contactToCheck : contactsWithErrorsList) {
+            System.assert(contactById.get(contactToCheck.Id) != null);
+        }
+    }
+
+    // Tests Prevent Deletion for Contacts with STUB_DoesNotHaveChildRelationships
+    @isTest
+    static void testPreventDeletionForContactsWithNoChildren() {
+
+        SRVC_Contact_PreventDeletion.instance = new STUB_DoesNotHaveChildRelationships();
+
+        Map<Id,Contact> contactById = new Map<Id,Contact>();
```

Comment:
Please add contacts to this map. Note that we do not need to insert them thanks to the magic of stubbing.